### PR TITLE
React/week2-Homework

### PR DESCRIPTION
**Pull request by islam-fawzy25**
// checkbox function to check if it is checked it will change status and make through line
  // i did my best here and still need help it is not working 100% i know and i am sure i have to useEffect here but i can not do more

### CODE REVIEW

File: react/week2/todo/src/App.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      </div>
+    );
+  };
+
+  // checkbox function to check if it is checked it will change status and make through line
+  // i did my best here and still need help it is not working 100% i know and i am sure i have to useEffect here but i can not do more 
+  
+  const changeTodoStatus = () => {
+setStatus(!status)
+  };
```
Comment (senner007) [inline react/week2/todo/src/App.js:74]: Do get the checkbox logic working, you simple have to apply the same approach as when you delete or add. Filter through the collection and change the todo status. You will need a status prop on each todo as well.

File: react/week2/todo/src/App.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  return (
+    <div>
+      <AddTodoButton />
+      <ul>
+        {array.map((obj) => (
```
Comment (senner007) [inline react/week2/todo/src/App.js:80]: I would prefer if you return a Todo component in your map function here. You "ListedTodos" is doing a lot of work here. Always look for ways to extract and separate your logic. A good indicator of too much responsibility is if you have large indentation landscapes as seen here.

Comment (senner007) [inline react/week2/todo/src/App.js:80]: Think about names here "array" and "obj" tend to add confusion and hide intention.

**Review by senner007 — APPROVED** [review]
Good.
You will soon get a hang of this.
Check the comments and see if you can add the checkbox logic

File: react/week2/todo/src/App.js (hunk 2 above)
Comment (islam-fawzy25) [reply react/week2/todo/src/App.js:80]: yes , i used meaningful words now

File: react/week2/todo/src/App.js (hunk 1 above)
Comment (senner007) [reply react/week2/todo/src/App.js:100]: Edit: Wrong phrasing. Filter is everyday speech. Sorry about the confusion. The above translates into using the map function here and returning a new object with the status property inverted where the id matches.

Comment (senner007) [reply react/week2/todo/src/App.js:100]: Here are examples of add, delete, and toggle methods:

```
 const add = () => {
    const getHigherId = Math.max(0, ...todos.map(todo => todo.id)) +1;  
    setTodos((todos) => {
      return [
        ...todos,
        {
          id: getHigherId,
          description: 'random text',
        },
      ];
    });

  };

  const remove = (id) => {
    setTodos((todos) => {
      return todos.filter((todo) => todo.id !== id);
    });
  };

  const toggleDone = (id) => {
    setTodos((todos) => {
      return todos.map((todo) =>
        todo.id === id ? { ...todo, isDone: !todo.isDone } : todo
      );
    });
  };
```

Comment (islam-fawzy25) [reply react/week2/todo/src/App.js:111]: checkbox now can change todo state when it checked but it changes the old todos only not the new one like random text

Comment (senner007) [reply react/week2/todo/src/App.js:111]: No the isDone is missing in the add function

Comment (islam-fawzy25) [reply react/week2/todo/src/App.js:111]: Thank you so much Niels for helping 
i know it was not the best code this time but Week3 i think I made it  better and I know more now about state and how to use it

**Review by senner007 — APPROVED** [review]

**Pull request merged**